    @Test
    public void testIntoAndOut() throws Exception {
        ArgumentCaptor<SuspendEvent> suspendEventCaptor = ArgumentCaptor.forClass(SuspendEvent.class);

        debugger.stepInto(new StepIntoActionImpl());

        verify(callback).onEvent(suspendEventCaptor.capture());
        SuspendEvent suspendEvent = suspendEventCaptor.getValue();
        assertEquals(suspendEvent.getLocation().getLineNumber(), 2);
        assertTrue(suspendEvent.getLocation().getTarget().endsWith("app.js"));

        Mockito.reset(callback);
        debugger.stepInto(new StepIntoActionImpl());

        verify(callback, timeout(1000)).onEvent(suspendEventCaptor.capture());
        suspendEvent = suspendEventCaptor.getValue();
        assertEquals(suspendEvent.getLocation().getLineNumber(), 5);
        assertTrue(suspendEvent.getLocation().getTarget().endsWith("app.js"));

        Mockito.reset(callback);
        debugger.stepOut(new StepOutActionImpl());

        verify(callback, timeout(1000)).onEvent(suspendEventCaptor.capture());
        suspendEvent = suspendEventCaptor.getValue();
        assertEquals(suspendEvent.getLocation().getLineNumber(), 9);
        assertTrue(suspendEvent.getLocation().getTarget().endsWith("app.js"));
    }

